HDPS/src/plugins/PointData/src/PointData.h

```diff
@@ -53,247 +54,129 @@ class ClusterAction;
-            {
-                return static_cast<ElementTypeSpecifier>(temp);
-            }
-            else
-            {
-                constexpr const std::tuple<Tail...>* tailNullptr{ nullptr };
-                return recursiveFindElementTypeSpecifier<ElementType>(tailNullptr, temp + 1);
-            }
-        }
-
-
-        template <typename ElementType>
-        constexpr static ElementTypeSpecifier recursiveFindElementTypeSpecifier(const std::tuple<>*, const int)
-        {
-            return ElementTypeSpecifier{}; // Should not occur!
-        }
-
-        template <typename ReturnType, typename VectorHolderType, typename FunctionObject, typename Head, typename... Tail>
-        static ReturnType recursiveVisit(VectorHolderType& vectorHolder, FunctionObject functionObject, const std::tuple<Head, Tail...>*)
-        {
-            using HeadValueType = typename Head::value_type;
-
-            if (vectorHolder.template isSameElementType<HeadValueType>())
-            {
-                return functionObject(vectorHolder.template getVector<HeadValueType>());
-            }
-            else
-            {
-                constexpr const std::tuple<Tail...>* tailNullptr{nullptr};
-                return recursiveVisit<ReturnType>(vectorHolder, functionObject, tailNullptr);
-            }
-        }
-
-        template <typename ReturnType, typename VectorHolderType, typename FunctionObject>
-        static ReturnType recursiveVisit(VectorHolderType&, FunctionObject&, const std::tuple<>*)
-        {
-            struct VisitException : std::exception
-            {
-                const char* what() const noexcept override
-                {
-                    return "visit error!";
-                }
-            };
-            throw VisitException{};
-        }
-
-    public:
-
-        /// Yields the n-th supported element type.
-        template <std::size_t N>
-        using ElementTypeAt = typename std::tuple_element_t<N, TupleOfVectors>::value_type;
-
-        /// Defaulted default-constructor. Ensures that the element type
-        /// specifier is default-initialized (to "float32").
-        VectorHolder() = default;
+public:
+    enum class ElementTypeSpecifier
+    {
+        float32,
+        bfloat16,
+        int16,
+        uint16,
+        int8,
+        uint8
+    };
 
-        /// Explicit constructor that copies the specified vector into the
-        /// internal data structure. Ensures that the element type specifier
-        /// matches the value type of the vector.
-        template <typename T>
-        explicit VectorHolder(const std::vector<T>& vec)
-            :
-            _elementTypeSpecifier{ getElementTypeSpecifier<T>() }
+private:
+    using VariantOfVectors = std::variant <
+        std::vector<float>,
+        std::vector<biovault::bfloat16_t>,
+        std::vector<std::int16_t>,
+        std::vector<std::uint16_t>,
+        std::vector<std::int8_t>,
+        std::vector<std::uint8_t> >;
+
```

Comment:
We should probably use this opportunity to extend the list with other standard types like double, std::int32_t, std::int64_t and std::uint32_t, std::uint64_t.

Reply:
@alxvth Thanks, I agree with you that this would be a good opportunity to adjust the type list (if necessary), although I would suggest to make that a separate commit, or even a separate pull request.

Reply:
Note that when the type list is adjusted, code for that uses one of the "visit" functions to support any type in the list should just keep working "automatically", whereas code that "switches" based on the type specifier should probably be adjusted manually. For example:

https://github.com/ManiVaultStudio/core/blob/8f8e5c9478349aff0fdc53bbcb1d9ca1aeba7aaf/HDPS/src/plugins/PointData/src/PointData.cpp#L113-L115 

No problem, just something to keep in mind!